doc/content.tex

```diff
@@ -1,1347 +0,0 @@
-
-    The monolithic architecture of the RHQ project was due its size hard to maintain and lacking robust REST API lead to
-    fresh development of new application. In contrast Hawkular consist of several loosely coupled or even independent
-    applications. These independent components are much easier to maintain and more importantly they communicate over
-    REST API.
-
-    This architecture of microservices and chosen protocol allow simple development of agents which can be
-    written in any programming language. In RHQ only Java agent were available.
-
-    Hawkular as a product is customized
-    Wildfly\footnote{An open source project of JBoss Enterprise Application Platform.} application server with all
-    components deployed in it.
-
-    List of Hawkular main components:
-    \begin{itemize}
-        \item Console\,--\,user web interface.
-        \item Accounts\,--\,authorization subsystem based on Keycloak\footnote{An open
-            source single sing-on and identity management for RESTful web services.}.
-        \item Inventory\,--\,graph based registry of all entities in Hawkular.
-        \item Metrics\,--\,time series metrics engine based on Cassandra\footnote{An open
-            source distributed database management system. Hybrid between key-value and
-        column-oriented database.}.
-        \item Alerts\,--\,alerting subsystem based on JBoss Drools.
-    \end{itemize}
-
-    Some of the modules also uses Java messaging system (JMS) for inter-component communication.
-
-    Modules are packaged as standard Java web archives (WAR), or enterprise archives (EAR) and deployed into Wildfly.
-    Build and package management is performed by Maven and Gulp for user interface modules.
-
-    %%%%%%%%
-    \section{Data Mining Goals} \label{sec:goals}
-    The goal of this thesis is to develop a module for Hawkular which will provide forecasts for any time series
-    metrics collected by agent.
-
-    On new metric data available the module learns from data and predicts new values. Based on this predicted values
-    an alert can be triggered. Forecast should be also available for user interface in predictive charts.
-
-    One Wildfly agent on average collects hundreds to thousands metrics, therefore module should be capable of
-    processing high volume of data. Some of the customers monitor hundreds of server each with multiple agents.
-    Therefore performance of chosen learning algorithm has to be taken in the account.
-
-    %%%%%%%%
-    \section{Metrics in Hawkular} \label{sec:metrics-in-hawkular}
-    In Hawkular there are three types of metrics: gauge, counter and availability. All of them are univariate metrics
-    of structure \\$\{timestamp, value\}$. Each of these types is used for collecting dedicated types of metric data.
-    For example gauge can increase or decrease over the time, counter is monotonically decreasing or increasing and
-    availability represents up or down state of a resource.
```

Comment:
For example gauge can increase or decrease over the time, counter  monotonically decreases or increases and
availability represents up or down state of a resource. (v tejto vete by bolo vhodne ponechat vsetko v jednoduchom pritomnom case, nakolko hovorime o ad1 faktoch, ad2 opakujuci sa proces